test/functional/fixtures/api/es-next/roles/test.js

```diff
@@ -0,0 +1,72 @@
+        return runTests('./testcafe-fixtures/configuration-test.js', 'Clear configuration', { shouldFail: true })
+            .catch(function (errs) {
+                expect(errs[0]).contains('- Error in Role initializer - A native alert dialog was invoked');
+                expect(errs[0]).contains('> 31 |    await t.click(showAlertBtn);');
+            });
+    });
+
+    it('Should restore configuration after role initializer', function () {
+        return runTests('./testcafe-fixtures/configuration-test.js', 'Restore configuration', { selectorTimeout: 5000 });
+    });
+
+    describe('Errors', function () {
+        it('Should fail all tests that use role with the initiliazer error', function () {
+            return runTests('./testcafe-fixtures/init-error-test.js', null, {
+                shouldFail: true,
+                only:       'chrome,ie,firefox'
+            })
+                .catch(function (errs) {
+                    const browsers = Object.keys(errs);
+
+                    expect(browsers.length).eql(3);
+
+                    browsers.forEach(function (browser) {
+                        expect(errs[browser].length).eql(2);
+
+                        errs[browser].forEach(function (err) {
+                            expect(err).contains('- Error in Role initializer - Error: Hey!');
+                            expect(err).contains('>  5 |    throw new Error(\'Hey!\');');
+                        });
+                    });
+
+                });
+        });
+
+        it('Should fail if role switched within initializer', function () {
+            return runTests('./testcafe-fixtures/errors-test.js', 'Role switch in initializer', { shouldFail: true })
+                .catch(function (errs) {
+                    expect(errs[0]).contains('- Error in Role initializer - Role cannot be switched while another role is being initialized.');
+                    expect(errs[0]).contains('> 4 |    await t.useRole(Role.anonymous());');
+                });
+        });
+
+        it('Should throw error if useRole argument is not a Role', function () {
+            return runTests('./testcafe-fixtures/errors-test.js', 'useRole argument', { shouldFail: true })
+                .catch(function (errs) {
+                    expect(errs[0]).contains('The "role" argument is expected to be a Role instance, but it was object.');
+                    expect(errs[0]).contains('> 16 |    await t.useRole({});');
+                });
+        });
+
+        it('Should fail if there error occurred while restoring configuration', function () {
+            return runTests('./testcafe-fixtures/errors-test.js', 'Error restoring configuration', { shouldFail: true })
+                .catch(function (errs) {
+                    expect(errs[0]).contains('- Error while restoring configuration after Role switch - The specified selector does not match any element in the DOM tree.');
```

Comment:
We have a [special error](https://github.com/DevExpress/testcafe/blob/2ecabd2a04a7411abf13839a3c6a90149d68ca00/src/errors/test-run/templates.js#L203) for cases when iframe does not exist anymore.
May be we should raise it in this case, wht do you think?